lib/protobuf/nats.rb

```diff
@@ -1,7 +1,49 @@
 require "protobuf/nats/version"
 
+require "protobuf"
+# We don't need this, but the CLI attempts to terminate.
+require "protobuf/rpc/service_directory"
+
+require "connection_pool"
+
+require "nats/io/client"
+
+require "protobuf/nats/client"
+require "protobuf/nats/server"
+require "protobuf/nats/runner"
+
 module Protobuf
   module Nats
-    # Your code goes here...
+    class << self
+      attr_accessor :client_nats_connection
+    end
+
+    module Messages
+      ACK = "\1".freeze
+    end
+
+    class Config
+      def self.connection_options
+        {:servers => ["nats://127.0.0.1:4222"]}
+      end
+    end
+
+    def self.subscription_key(service_klass, service_method)
+      service_class_name = service_klass.name.underscore.gsub("/", ".")
+      service_method_name = service_method.to_s.underscore
+      "rpc.#{service_class_name}.#{service_method_name}"
+    end
+
+    def self.start_client_nats_connection
+      @client_nats_connection = ::NATS::IO::Client.new
+      @client_nats_connection.connect(Config.connection_options)
+
+      # Ensure we have a valid connection to the NATS server.
+      @client_nats_connection.flush(5)
+
+      true
+    end
+
+    start_client_nats_connection
```

Comment:
This is actually a connection used only by the rpc client. I agree it's not the best solution and we've had issues managing this with action subscriber, but I'm all 👂 s for a better method. Maybe we should push this to the library user? They need to configure the client connection similar to sidekiq?